app/src/pages/recipes/create.tsx

```diff
@@ -0,0 +1,175 @@
+    const [loading, setLoading] = useState(true);
+    const [requestCount, setRequestCount] = useState(0);
+    const {handleError} = useHandleError();
+    const addToast = useToast();
+
+    const handleGetIngredients = (data: IngredientType[]): void => {
+        setIngredients(data);
+        setRequestCount((prevState) => prevState + 1);
+    }
+
+    const handleGetAuthors = (data: AuthorType[]): void => {
+        setAuthors(data);
+        setRequestCount((prevState) => prevState + 1);
+    }
+
+    useEffect(() => {
+        if (!loading) {
+            return;
+        }
+
+        indexIngredients(
+            (data: IngredientType[]) => handleGetIngredients(data),
+            (errorCode: number) => handleError(errorCode)
+        );
+
+        indexUsers(
+            (data: AuthorType[]) => handleGetAuthors(data),
+            (errorCode: number) => handleError(errorCode)
+        );
+    }, [loading]);
+
+    useEffect(() => {
+        // since we have two separate requests, we need to wait for both to finish
+        // so keep track of them independently via counter. When it's 2, we're ready.
+
+        // if you can think of a better way of programming this, please let me know
+        // ps sorry for the setTimeout, I'm not proud of it either
+        if (requestCount === 2) {
+            setTimeout(() => setLoading(false), 500);
+        }
+    }, [requestCount]);
+
+    const handleSelectAuthor = (setFieldValue: Function, selectedOption: OptionType): void => {
+        // ensure Formik knows that we've selected an author in our custom select component
+        setFieldValue('author_id', selectedOption.value);
+    };
+
+    const handleSelectIngredient = (setFieldValue: Function, selectedOptions: OptionType[]): void => {
+        // remap the keys since our select component expects a specific shape different to our API payload
+        let selectedIngredients = fromSelectOptions(selectedOptions);
+
+        // ensure Formik knows that we've selected a couple options in our custom select component
+        setFieldValue('ingredients', selectedIngredients);
+    };
+
+    if (loading) {
+        return <Spinner type={'primary'} size={'medium'}
+                        text={<FormattedText>{fun.loadingOne.defaultMessage}</FormattedText>}/>;
+    }
+
+    const CardBodySpacer = styled.div`
+        padding: 24px;
+        width: 50%;
+    `
+
+    const FieldSet = styled.div`
+        margin: 1em 0;
+    `
+
+    /**
+     * Quick note on Formik:
+     * Offloads most of the pain of form handling to the library.
+     * It stores form state via formik.values
+     * For the select component I've had to 'step out' of formik because it's multi-select
+     * isn't great.
+     * Use the CreateRecipeSchema to validate the form.
+     * If the form doesn't submit, it's bc values from onSubmit do not match the Schema.
+     */
+
+    return (
+        <div>
+
+            <Flex flexDirection={'column'}>
```

Comment:
I never new that we had a `Flex` component. I will need to look and see what this does but this seems so pointless to me. <- Having a go a suitcase being too hand holding not at the code.